packages/babel-preset-typescript/src/plugin-rewrite-ts-imports.ts

```diff
@@ -14,12 +24,18 @@ export default declare(function ({ types: t }) {
         const kind = t.isImportDeclaration(node)
           ? node.importKind
           : node.exportKind;
-        if (kind === "value" && source && /[\\/]/.test(source.value)) {
-          source.value = source.value
-            .replace(/(\.[mc]?)ts$/, "$1js")
-            .replace(/\.tsx$/, ".js");
+        if (kind === "value" && source) {
+          maybeReplace(source);
         }
       },
+      CallExpression(path) {
+        if (t.isImport(path.node.callee)) {
+          maybeReplace(path.node.arguments[0]);
+        }
+      },
+      ImportExpression(path) {
+        maybeReplace(path.node.source);
```

Comment:
I saw that the TS team was also considering using a runtime helper to replace the extension in cases where the specifier is not statically analyzable. We should consider doing something like that.